examples/golang/build.go

```diff
@@ -74,15 +88,45 @@ func BuildContainer() (err error) {
 
 // Test is an example mage target that runs the go compilers tests
 // Note: currently assumes project extracted to a GOPATH directory structure
-func Test() (err error) {
+func Test(ctx context.Context) (err error) {
 	t := root.Target("test")
 	defer func() { t.Complete(err) }()
 
 	err = t.GolangTest().
-		SetBuildContainer("golang:1.14").
-		Test(context.TODO(), "github.com/gravitational/magnet/examples/golang")
+		SetEnv("GO111MODULE", "on").
+		SetBuildContainer("builder:1.14").
+		Test(ctx, "github.com/gravitational/magnet/examples/golang")
 	if err != nil {
 		return trace.Wrap(err)
 	}
 	return
 }
+
+// BuildContainer creates a docker container as a consistent Go environment to use for software builds.
+func BuildContainer(ctx context.Context) (err error) {
+	m := root.Target("buildContainer")
+	defer func() { m.Complete(err) }()
+
+	return m.DockerBuild().
+		AddTag("builder:1.14").
```

Comment:
Also, instead of hardcode "builder:1.14" I think this would probably be a variable if not an example, so might be worth having the example the way we'd want it in a real project.

Reply:
Addressed in https://github.com/gravitational/magnet/pull/2/commits/fa3aca08010b89e4bc5351a34cab37f248ff6fbc